src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { Navigate, Outlet, useOutletContext } from "react-router-dom";
+import { User } from "../api";
+
+interface Props {
+	user: User | null;
+	redirectPath: string;
+}
+
+type ProtectedRouteProps = { user: User };
+
+export function ProtectRoute({ user, redirectPath }: Props) {
```

Comment:
Now that this is in place, we probably want to scan the codebase to remove all `User | null` where its no longer needed, and use the new `getUser` function where we previously used the old one

Reply:
I think with fixing my merge conflicts of the main after the other two PRs got merged I call all the `User | null` that were suppose to be there any more. I think the only place I see it now is in places where the `protectedRoute` isn't used like the `App.tsx`, `Home.tsx`, `Layout.tsx` and in `protectedRoute`.